packages/vite/src/index.ts

```diff
@@ -126,10 +126,12 @@ export function imagetools(userOptions: Partial<VitePluginOptions> = {}): Plugin
         const { transforms } = generateTransforms(config, transformFactories, srcURL.searchParams, logger)
         const { image, metadata } = await applyTransforms(transforms, img.clone(), pluginOptions.removeMetadata)
 
-        if (viteConfig.command === 'serve') {
+        if (directives.has('inline')) {
+          metadata.src = `data:image/${metadata.format};base64,${(await image.toBuffer()).toString('base64')}`
+        } else if (viteConfig.command === 'serve') {
           const id = await generateImageID(srcURL, config, img)
           generatedImages.set(id, image)
```

Comment:
I think we'd still want to do this portion if `inline` and `serve` are both true so that we don't have to recompute the image on subsequent requests. maybe we could do something like:
```
let id;
if (viteConfig.command === 'serve') {
          id = await generateImageID(srcURL, config, img)
          generatedImages.set(id, image)
}
if (directives.has('inline')) {
          metadata.src = `data:image/${metadata.format};base64,${(await image.toBuffer()).toString('base64')}`
} else if (viteConfig.command === 'serve') {
         metadata.src = join(viteConfig?.server?.origin ?? '', basePath) + id
} else {
...
}
```

Reply:
Updated